src/util/Math.cpp

```diff
@@ -159,25 +159,41 @@ k_means(std::vector<double> const& points, uint32_t k)
     return centroids;
 }
 
-#ifdef BUILD_TESTS
 static unsigned int lastGlobalSeed{0};
-void
-reinitializeAllGlobalStateWithSeed(unsigned int seed)
+#ifndef BUILD_TESTS
+static
+#endif
+    void
+    reinitializeAllGlobalStateWithSeed(unsigned int seed)
 {
     lastGlobalSeed = seed;
     PubKeyUtils::clearVerifySigCache();
     srand(seed);
     gRandomEngine.seed(seed);
+#ifdef BUILD_TESTS
```

Comment:
up to you, but given that test vs prod build randomness are sufficiently different, it seems cleaner/more readable to just have two different functions (one behind BUILD_TESTS flag) that initialize randomness.

Reply:
good point. fixed